aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -117,9 +120,39 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (exceptionThrown) {
         io().deleteFile(newMetadataLocation);
       }
+
+      unlock();
```

Comment:
Can you put this in a separate try/finally block?

If any of the exceptions above is thrown, then this needs to delete the metadata location. But if that for some reason fails, this also needs to attempt to unlock. Since both unlock and deleteFile must be done even if the other fails, we need to separate them into different finally blocks that are independent.

Reply:
Interesting, this is the same logic as `HiveTableOperations`, I thought about this issue and explained to myself that we will just wait for the lock to timeout in that case. We should probably do the same in Hive then.